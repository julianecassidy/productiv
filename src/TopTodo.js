import React from "react";

import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - toDos
 *
 * TodoApp -> TopTodo
 */

function TopTodo({ toDos }) {
  // lowest-priority # is the highest priority
  let top = toDos.reduce(
    (acc, cur) => (cur.priority < acc.priority ? cur : acc),
    toDos[0]
  );
  console.log(top, "TOP TODO");

  return (
    <Todo
      id={top.id}
      title={top.title}
      description={top.description}
      priority={top.priority}
    />
  );
}

// id, title, description, priority

export default TopTodo;
