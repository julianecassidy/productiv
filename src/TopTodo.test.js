import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

const  TEST_TODO_0 = {
  title: "test todo priority 2",
  description: "test desc",
  priority: 2,
  id: 0
};

const TEST_TODO_1 = {
  title: "test todo priority 1",
  description: "test desc",
  priority: 1,
  id: 1,
};

const TEST_TODO_2 = {
  title: "test todo priority 1 2nd version",
  description: "test desc",
  priority: 1,
  id: 2,
};

const TEST_TODO_3 = {
  title: "test todo priority 3",
  description: "test desc",
  priority: 3,
  id: 3,
};

const TEST_TODOS = [TEST_TODO_0, TEST_TODO_1, TEST_TODO_2, TEST_TODO_3];

describe("TopTodo component", function () {
  it("renders without crashing", function () {
    render(<TopTodo toDos={TEST_TODOS}/>);
  });

  it("matches snapshot", function () {
    const { container } = render(
      <TopTodo toDos={TEST_TODOS} />
    );
    expect(container).toMatchSnapshot();
  });

  it("Renders the correct Todo", function () {
    const { container } = render(
      <TopTodo toDos={TEST_TODOS} />
    );

    expect(container).toContainHTML("test desc");
    expect(container).toContainHTML("Priority: 1");
    expect(container).toContainHTML("test todo priority 1");
  });

  it("If no priority 1, renders next lowest Todo", function () {
    const { container } = render(
      <TopTodo toDos={[TEST_TODO_3]} />
    );

    expect(container).toContainHTML("test desc");
    expect(container).toContainHTML("Priority: 3");
    expect(container).toContainHTML("test todo priority 3");  
  });

});