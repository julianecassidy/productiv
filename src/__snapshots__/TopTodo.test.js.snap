// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo component matches snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
        test todo priority 1
      </b>
       
      <small>
        Priority: 
        1
      </small>
    </div>
    <div>
      <small>
        test desc
      </small>
    </div>
  </div>
</div>
`;
